core/src/main/java/tech/tablesaw/table/TableSlice.java

```diff
@@ -184,4 +242,53 @@ public boolean hasNext() {
       }
     };
   }
+
+  private boolean hasSelection() {
+    return selection != null;
+  }
+
+  private boolean isSorted() {
+    return sortOrder != null;
+  }
+
+  /**
+   * Maps the view row number to the row number on the underlying source table.
+   *
+   * @param rowNumber the row number in the view.
+   * @return the matching row number in the underlying table.
+   */
+  public int mappedRowNumber(int rowNumber) {
+    if (isSorted()) {
+      return sortOrder[rowNumber];
+    } else if (hasSelection()) {
+      return selection.get(rowNumber);
+    }
+    return rowNumber;
+  }
+
+  /**
+   * Sort the TableSlice independently from the underlying table.
+   */
+  public void sortOn(Sort key) {
```

Comment:
It looks like this duplicates a method in `Table`. If this method is moved to `Relation` then we don't need it in `Table` or `TableSlice`

Reply:
I agree this is not well documented.
`Table#sortOn` returns a sorted copy of the Table. `TableSlice#sortOn` returns void and sorts the slice/view in place. I will send a PR improving the javadoc.